introspection-engine/connectors/sql-introspection-connector/src/introspection.rs

```diff
@@ -239,3 +230,40 @@ fn calculate_fields_for_prisma_join_table(
         }
     }
 }
+
+fn introspect_enums(datamodel: &mut Datamodel, ctx: &Context) {
+    let existing_enums_by_database_name: HashMap<&str, _> = ctx
+        .previous_schema()
+        .db
+        .walk_enums()
+        .map(|enm| (enm.database_name(), enm.id))
+        .collect();
+
+    datamodel.enums = ctx
+        .schema
+        .enum_walkers()
+        .map(|enm| sql_enum_to_dml_enum(enm, ctx))
+        .collect();
+
+    datamodel.enums.sort_by(|a, b| {
+        let existing = |enm: &dml::Enum| -> Option<_> {
+            existing_enums_by_database_name.get(enm.database_name.as_deref().unwrap_or(enm.name.as_str()))
+        };
+        match (existing(a), existing(b)) {
+            (Some(a), Some(b)) => a.cmp(b),
+            (Some(_), None) => std::cmp::Ordering::Less,
+            (None, Some(_)) => std::cmp::Ordering::Greater,
+            (None, None) => std::cmp::Ordering::Equal,
+        }
```

Comment:
We need this because the implementation of `PartialOrd` for options puts `None`s first, but we want `None`s last.

Reply:
We may want to move it to its own function though, we need it in multiple places in introspection. I'll look into that real quick.

Reply:
Done.